src/highlightjs/src/bicep.ts

```diff
@@ -43,7 +43,7 @@ const KEYWORDS = {
 
 const lineComment: Mode = {
   className: 'comment',
-  match: `//.*$`,
+  match: `//.*${before(`$`)}`,
```

Comment:
I believe this is the heart of the problem - matches for objectProperty and comment expect to be able to capture `$`, but only the inner pattern succeeds - messing up the other match.